### PR TITLE
[tools] Move logic to validate/initialize DeploymentTarget to shared code.

### PR DESCRIPTION
**Pull request by rolfbjarne**
So that it's executed for .NET as well.

### CODE REVIEW

**Review by spouliot — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: tools/common/Application.cs [hunk 1]
```diff
@@ -794,6 +794,27 @@ public void InitializeCommon ()
...
 			}
 
+
+#if ENABLE_BITCODE_ON_IOS
+			if (Platform == ApplePlatform.iOS)
+				DeploymentTarget = new Version (9, 0);
+#endif
+
+			if (DeploymentTarget == null)
+				DeploymentTarget = Xamarin.SdkVersions.GetVersion (this);
```
Comment (mandel-macaque) [inline tools/common/Application.cs:804]: not a needed change, but you can do ??= 
```suggestion
				DeploymentTarget ??= Xamarin.SdkVersions.GetVersion (this);
```

**Review by mandel-macaque — APPROVED** [review]

**Comment by mandel-macaque** [comment]
VSTS could not checkout the code :/

**Comment by rolfbjarne** [comment]
Test failure is unrelated (https://github.com/xamarin/xamarin-macios/issues/10236).

**Pull request merged**